scripts/utils.mjs

```diff
@@ -1,11 +1,33 @@
 import path from 'path';
 import url from 'url';
+import fs from 'fs';
+
+function findUpFile(fileName, cwd = process.cwd(), maxIterations = 5) {
```

Comment:
Previously I was using `find-up` package but later realised that this won't work with the `useReactVersion` script if we were to import `getWorkspaceRoot` in that file since it runs before `pnpm install`.

Reply:
But `useReactVersion` from core repo doesn't use `getWorkspaceRoot`, so maybe these changes aren't necessary anymore?

I'd like to eventually get rid of `getWorkspaceRoot`, there doesn't seem to be any case where this is strictly needed. It makes those scripts harder to port. We should just require users to pass a path if they want to operate on a different folder than CWD.

But this can be done in follow-up.

Reply:
True. But its required in other scripts. We can't use it in `useReactVersion` anyway because there won't be any packages installed when the script runs.
Base UI was using their version of `getWorkspaceRoot` in the `useReactVersion` script. So I though I'd update it in core as well.
But reverted later since it won't work across repos.